addOns/paramminer/src/main/java/org/zaproxy/addon/paramminer/UrlBruteForce.java

```diff
@@ -0,0 +1,321 @@
+        }
+        ParamReasons res = compare(base, response, params);
+        if (mode == Mode.VERIFY) {
+            if (res != null) {
+                for (String parameter : res.getParams().keySet()) {
+                    ParamGuessResult paramGuessResult =
+                            new ParamGuessResult(parameter, res.getReason(), msg);
+                    guessedParams.add(paramGuessResult);
+                }
+            }
+        }
+        return res;
+    }
+
+    /**
+     * Checks ths status codes of the response for succesful requests.
+     *
+     * @param base the base response
+     * @param response the response to check
+     * @return Status.OK if the response is valid, false otherwise.
+     */
+    public Status errorHandler(ComparableResponse base, ComparableResponse response) {
+        int status = response.getStatusCode();
+        if (ERRORCODES.contains(status)) {
+            if (status == 503) {
+                // TODO Display on out panel "Taget unable to process requests"
+                return Status.KILL;
+            } else if (status == 429 || status == 418) {
+                // TODO Display on out panel "Target is rate limited"
+                return Status.KILL;
+            } else {
+                if (base.getStatusCode() != response.getStatusCode()) {
+                    return Status.KILL;
+                } else {
+                    return Status.OK;
+                }
+            }
+        } else if (response.getBody().isEmpty() || response.getHeaders().size() == 0) {
+            return Status.KILL;
+        }
+        return Status.OK;
+    }
+
+    public String requester(HttpMessage msg, Method method, Map<String, String> params) {
+        ParamMinerHistoryTableModel table;
+        if (method.equals(Method.GET)) {
+            try {
+                table = scan.getTableModel();
+                // TODO Add right options so that http message can be loaded using right click menu.
+                String uri = config.getUrl();
+                String queryString = UrlUtils.createQueryString(params);
+                HttpRequestHeader headers = new HttpRequestHeader(uri);
+                headers.setMethod(HttpRequestHeader.GET);
+                headers.setURI(new URI(uri + queryString, true));
+                for (HttpHeaderField header : msg.getRequestHeader().getHeaders()) {
+                    headers.setHeader(header.getName(), header.getValue());
+                }
+                msg.setRequestHeader(headers);
+                httpSender.sendAndReceive(msg);
+                table.addHistoryReference(
+                        new HistoryReference(Model.getSingleton().getSession(), 23, msg));
+                return StringUtils.strip(queryString, "?");
+            } catch (Exception e) {
+                // TODO show proper error message on Output Panel (can be timeout or connection
+                // refused)
+                e.printStackTrace();
```

Comment:
Better to use logging rather than `e.printStackTrace()` and hardly any more work 😉 Same for other instances..

Reply:
Okay. The output section would be added in the next PRs